### PR TITLE
Feat/integ 816

### PR DESCRIPTION
**Pull request by rusticpenguin**
## Purpose

This PR adds full feature parity with AiCG V1

What is not in this PR? 
1. The styling does not quite match the designs found [here](https://www.figma.com/file/TQsQcvECzI6R2Jy2Utu6xP/AI-Content-Generator-Revision-2?type=design&node-id=118-59938&mode=design&t=yzbFs4saYqB6imN3-0). This will be accomplished in a follow-up ticket

### CODE REVIEW

File: apps/ai-content-generator/src/locations/Dialog.tsx [hunk 1]
```diff
@@ -11,24 +12,24 @@ const Dialog = () => {
   const { feature, entryId, isLoading } = useDialogParameters();
 
   if (isLoading) {
-    return <Spinner size="large" />;
+    return null;
```
Comment (rusticpenguin) [inline apps/ai-content-generator/src/locations/Dialog.tsx:15]: Just switched this to null since the spinner is never visible

File: apps/ai-content-generator/src/components/dialog/common-generator/generatorReducer.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+export enum GeneratorAction {
```
Comment (jsdalton) [inline apps/ai-content-generator/src/components/dialog/common-generator/generatorReducer.tsx:1]: For me these action names are a bit confusing. The first few map to an event (with verb) but the latter are just nouns. Is there a reason why they are not named as a typical verb actions? We might consider renaming for clarity (makes it easy to see what event actually triggers the action and associated state change)

File: apps/ai-content-generator/src/components/dialog/common-generator/field-selector/field-list/EntryFieldList.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    </Select.Option>
+  ));
+
+  return (
+    <FormControl>
+      <FormControl.Label>{title}</FormControl.Label>
+      <Select value={selectedField} onChange={onChange}>
+        <Select.Option value="" isDisabled>
+          Select field...
+        </Select.Option>
```
Comment (lilbitner) [inline apps/ai-content-generator/src/components/dialog/common-generator/field-selector/field-list/EntryFieldList.tsx:27]: it looks to me when playing around with the branch that when the 'Select field...' is displayed, a user can still try and generate text. I think this might be remedied by a flow in Emily's designs ([example](https://www.figma.com/file/TQsQcvECzI6R2Jy2Utu6xP/AI-Content-Generator-Revision-2?type=design&node-id=176-66812&mode=design&t=YyulQ5JWpigk4fFj-4))

File: apps/ai-content-generator/src/components/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        <TextFieldWithButtons inputText={output} onFieldChange={handleGeneratedTextChange}>
+          <>
+            <CopyButton value={output} />
+            <Button onClick={generate}>Regenerate</Button>
+            <Button isDisabled={hasOutputField} onClick={apply}>
+              Apply
+            </Button>
+          </>
+        </TextFieldWithButtons>
+      )}
```
Comment (lilbitner) [inline apps/ai-content-generator/src/components/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.tsx:39]: nit - could either way here around if a ternary is more readable.` isGenerating ? ___ : ___`

File: apps/ai-content-generator/src/hooks/dialog/useEntryAndContentType.ts [hunk 5]
```diff
@@ -25,13 +26,40 @@ const useEntryAndContentType = (entryId: string) => {
...
+
+    return { entry, contentType };
+  };
+
+  const updateEntry = async (fieldKey: string, fieldLocale: string, updatedText: string) => {
+    try {
+      const { entry, contentType } = await getEntryAndContentType();
+      if (entry === null || contentType === null) {
+        throw new Error('Entry or content type is null');
+      }
```
Comment (lilbitner) [inline apps/ai-content-generator/src/hooks/dialog/useEntryAndContentType.ts:38]: how is this error being handled in terms of user facing actionable issues? I am assuming if this error is thrown something went wrong, and the user will not be able to take action to remedy?

File: apps/ai-content-generator/src/hooks/dialog/useSupportedFields.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+      );
+
+      return validatedFields;
+    }
+
+    return { supportedFields: [], fields: [] };
+  }, [entry, contentType, targetLocale, supportedFields]);
+
+  return { ...fields };
+};
```
Comment (lilbitner) [inline apps/ai-content-generator/src/hooks/dialog/useSupportedFields.ts:63]: nice use of `useMemo` 🔥

File: apps/ai-content-generator/src/locations/Dialog.tsx (hunk 1 above)
Comment (lilbitner) [reply apps/ai-content-generator/src/locations/Dialog.tsx:15]: Does this state need to exist if we just render `null`? or is it accounting for a split second of loading?

**Review by jsdalton — COMMENTED** [review]
Looking good! I left one comment and I think it's more of an overall thing. I found it a bit hard to follow the actions and effects here because of the action names and concepts being a bit arbitrary. My thought would be if the action types were more clear (describe what happened) it would be easier to follow.

File: apps/ai-content-generator/src/providers/dialog/common-generator/generatorProvider.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+  };
+
+  return (
+    <GeneratorContext.Provider value={{ ...providerData, setProviderData: updateProviderData }}>
+      {children}
+    </GeneratorContext.Provider>
+  );
+};
+
+export default GeneratorProvider;
```
Comment (lilbitner) [inline apps/ai-content-generator/src/providers/dialog/common-generator/generatorProvider.tsx:57]: nit: does this file need to be nested so far within the `providers` directory? Can all providers exist at the same level within the directory?

File: apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+    const isSupported = supportedFields.includes(field.type as SupportedFieldTypes);
+    if (isSupported) {
+      fieldAcc.supportedFields.push(formattedField);
+    }
+
+    fieldAcc.fields.push(formattedField);
+    return fieldAcc;
+  };
+};
```
Comment (lilbitner) [inline apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts:35]: nit: I think it might be helpful, as you do with the hooks, to add a small description around what these helpers do.

**Review by lilbitner — COMMENTED** [review]
Working really well George! ⭐  just a few comment!

**Review by whitelisab — COMMENTED** [review]
Looks awesome! ⭐ I left some comments that can either be addressed on this PR or in follow up work.

File: apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+  if (field.type === 'RichText') {
+    formattedField.data = documentToPlainTextString(formattedField.data);
+  }
+
+  return formattedField;
+};
+
+const isSupported = (entry: EntryProps, supportedFields: SupportedFieldTypes[], locale: string) => {
+  return (fieldAcc: SupportedFieldsOutput, field: ContentFields) => {
+    const existsInEntry = entry.fields[field.id];
```
[the remark this answers is not in the document]
Comment (rusticpenguin) [reply apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts:20]: Good catch! The first should be an easy fix. I'll go fix this up

File: apps/ai-content-generator/src/components/dialog/common-generator/generatorReducer.tsx (hunk 2 above)
Comment (rusticpenguin) [reply apps/ai-content-generator/src/components/dialog/common-generator/generatorReducer.tsx:1]: Good catch! I'm going to add verbs to these as well as the actions for the Config page

File: apps/ai-content-generator/src/hooks/dialog/useEntryAndContentType.ts [hunk 10]
```diff
@@ -25,13 +26,40 @@ const useEntryAndContentType = (entryId: string) => {
...
+      const fieldType = contentType?.fields.find((field) => field.id === fieldKey)?.type;
+      const isRichText = fieldType === 'RichText';
+
+      entry.fields[fieldKey][fieldLocale] = isRichText ? richTextModel(updatedText) : updatedText;
+
+      await cma.entry.update(
```
[the remark this answers is not in the document]
Comment (whitelisab) [reply apps/ai-content-generator/src/hooks/dialog/useEntryAndContentType.ts:44]: Awesome, thanks for clarifying re: the dialog location not having access to `sdk.field`!

File: apps/ai-content-generator/src/components/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.tsx (hunk 4 above)
Comment (rusticpenguin) [reply apps/ai-content-generator/src/components/dialog/common-generator/output/output-text-panels/generated-text-panel/GeneratedTextPanel.tsx:39]: I can go either way on this too. Let's use the ternary!

File: apps/ai-content-generator/src/components/dialog/common-generator/field-selector/available-locales/AvailableLocales.tsx [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+const AvailableLocale = (props: Props) => {
+  const { title, selectedLocale, onChange } = props;
+  const sdk = useSDK<DialogAppSDK>();
+
+  const localeList = sdk.locales.available.map((locale) => {
+    return (
+      <Select.Option key={locale} value={locale}>
+        {locale}
```
[the remark this answers is not in the document]
Comment (rusticpenguin) [reply apps/ai-content-generator/src/components/dialog/common-generator/field-selector/available-locales/AvailableLocales.tsx:19]: Fixed!

File: apps/ai-content-generator/src/components/dialog/common-generator/field-selector/field-list/EntryFieldList.tsx (hunk 3 above)
Comment (rusticpenguin) [reply apps/ai-content-generator/src/components/dialog/common-generator/field-selector/field-list/EntryFieldList.tsx:27]: Created a ticket on this!

File: apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts (hunk 8 above)
Comment (rusticpenguin) [reply apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts:35]: Added small descriptions to all the helpers! Really appreciate the attention to quality here

**Comment by rusticpenguin** [comment]
> Oh and I had one more comment about the file structure. I think we are following a nice pattern in GA4 and Ecomm where the components are broken down first into directories of _App, Config, or Common_, and then further broken down like you have them (Dailog, Sidebar etc). I find that makes it a lot easier to find things within the files. What do you think?

I forgot I structure Ecomm's frontend like that! My bad on consistency there! Thanks for catching that, I'll move them into a /app folder to keep with the style

File: apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts (hunk 8 above)
Comment (lilbitner) [reply apps/ai-content-generator/src/utils/dialog/supported-fields/supportedFieldsHelpers.ts:35]: Thanks George!! 🙌

**Review by lilbitner — APPROVED** [review]
Looks great 👍 ✨
